segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/IdempotentStorageTestBase.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage;
+
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.test.common.AssertExtensions;
+import java.io.ByteArrayInputStream;
+import java.util.concurrent.CompletableFuture;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static io.pravega.test.common.AssertExtensions.assertMayThrow;
+import static io.pravega.test.common.AssertExtensions.assertSuppliedFutureThrows;
+
+/**
+ * Common Unit tests for FileSystemStorage and ExtendedS3Storage.
```

Comment:
While we are at it, let's also fix the comment w. 
Should read ` Common Unit tests for Storage implementations that allow idempotent writes.`

Reply:
Yes Sachin